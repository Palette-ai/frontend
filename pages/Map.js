import React from 'react';
import { SafeAreaView, StyleSheet } from 'react-native';
import MapView, { PROVIDER_GOOGLE } from 'react-native-maps';

	return (
		<SafeAreaView style={styles.container}>
			<MapView
				style={{ flex: 1 }}
				provider={'google'}
				showsUserLocation={true}
				initialRegion={{
					latitude: 43.705130,
					longitude: -72.289520,
					latitudeDelta: .0095,
					longitudeDelta: .0095
				}}
			>
				{data.restaurantMany.map(r => {
					return (

						<Marker
							key={r._id}
							coordinate={{
								latitude: r.latitude,
								longitude: r.longitude
							}}
							title={r.name}
							description={r.phone_number}
						>
							<Callout
								onPress={() => navigation.navigate('Restaurants', { r, navigation })}
							>
								<View>
									<Text>{r.name}</Text>
									<Text>{r.phone_number}</Text>
								</View>
							</Callout>
							{/* <View style={{ backgroundColor: "red", padding: 10 }}>
								<Text>{r.name}</Text>
							</View> */}
							{/* <View /> */}
						</Marker>
					)
				})}
			</MapView>
		</SafeAreaView>
	)
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
	},
});

export default Map
