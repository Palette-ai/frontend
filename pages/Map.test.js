import React from 'react';
import Map from './Map';
import {shallow} from 'enzyme'
import "../setupTests"

const map = shallow(<Map />);

describe('renders correctly', () => {
  it('should render the Map Component correctly', () => {   
    expect(map).toMatchSnapshot();
  });
});
