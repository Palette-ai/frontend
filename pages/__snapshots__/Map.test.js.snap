// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly should render the Map Component correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "flex": 1,
    }
  }
>
  <MapView
    initialRegion={
      Object {
        "latitude": 43.70513,
        "latitudeDelta": 0.0095,
        "longitude": -72.28952,
        "longitudeDelta": 0.0095,
      }
    }
    provider="google"
    showsUserLocation={true}
    style={
      Object {
        "flex": 1,
      }
    }
  />
</ForwardRef(SafeAreaView)>
`;
